scripts/darken_colors.lua

```diff
@@ -0,0 +1,143 @@
+-- Plugin lifepillar/vim-colortemplate must be available on &runtimepath
+--
+-- The current working directory must be set to the repo root
+-- Have the main file already open in another window
+
+-- IMPORTANT: the `cterm_color` key must always be below the `color` key of the
+--            same icon. Currently they are.
+
+-- After this file is sourced, the newly created file should open in the same
+-- window of the main file.
+
+-- :source this file when ready
+
+local fn = vim.fn
+
+if not jit then
+  error("Neovim must be LuaJIT-enabled to source this script")
+end
+
+if fn.filereadable("lua/nvim-web-devicons.lua") == 0 then
+  error("Your working directory isn't set to correctly.")
+end
+
+local light78 = 255 * 21 / 8 -- (255 * 3) * 7 / 8
+local light34 = 255 * 9 / 4 -- (255 * 3) * 3 / 4
+local light12 = 255 * 3 / 2 -- (255 * 3) / 2
+local light13 = 255         -- (255 * 3) / 3
+
+local function darken_color(rrggbb)
+  local r, g, b = rrggbb:match"%#(%x%x)(%x%x)(%x%x)"
+  r, g, b = tonumber("0x" .. r), tonumber("0x" .. g), tonumber("0x" .. b)
+  local lum = r + g + b
+  if lum < light13 then -------------------- in the darkest tertile already
+    return rrggbb
+  elseif lum < light12 then ---------------- darker than average
+    r = bit.tohex(r / 4 * 3):sub(-2)
+    g = bit.tohex(g / 4 * 3):sub(-2)
+    b = bit.tohex(b / 4 * 3):sub(-2)
+  elseif lum < light34 then ---------------- second lightest quartile
+    r = bit.tohex(r / 3 * 2):sub(-2)
+    g = bit.tohex(g / 3 * 2):sub(-2)
+    b = bit.tohex(b / 3 * 2):sub(-2)
+  elseif lum < light78 then ---------------- second lightest octile
+    r = bit.tohex(r / 3):sub(-2)
+    g = bit.tohex(g / 3):sub(-2)
+    b = bit.tohex(b / 3):sub(-2)
+  else ------------------------------------- lightest octile
+    r = bit.tohex(r / 5):sub(-2)
+    g = bit.tohex(g / 5):sub(-2)
+    b = bit.tohex(b / 5):sub(-2)
+  end
+  return string.format("#%s%s%s", r, g, b)
+end
```

Comment:
Could we perhaps go a bit darker? This looks good (base16-default-light with #ffffff background) however some are still a little indistinct:
![20230304_142722](https://user-images.githubusercontent.com/1505378/222873573-e7f97fd0-ab65-4d29-9a0b-5df1a4e08df8.png)

Reply:
You mean the folder icon or which ones?

Reply:
I was thinking about config.mk. The folder ones can be ignored as they are from nvim-tree.

Don't worry for now; I might run a test with all the icons and see how they all look.